resqpy/lines/_polyline.py

```diff
@@ -548,12 +548,13 @@ def splined(self,
         if rep_int_root is None:
             rep_int_root = self.rep_int_root  # todo: check whether it is legal to have 2 representations for 1 interpretation
 
-        return Polyline(self.model,
-                        set_bool = self.isclosed,
-                        set_coord = spline_coords,
-                        set_crs = self.crs_uuid,
-                        title = title,
-                        rep_int_root = rep_int_root)
+        cls = type(self)  # Get class constructor object i.e. Polyline
```

Comment:
This is an example of a code change which I regard as logical but if anything making the code harder to read. If the class name ever changes in future, it would be a significant breaking change requiring a sweep of the code base anyway, so having one less change to make would not save much in that unlikely event. Having said all that, I don't feel strongly enough about it to request the change is undone.

Reply:
Ah yes. I changed that whilst debugging, as I was experimenting changing the class name to `_Polyline` and the in the init using `import _Polyline as Polyline`, and saw that change broke the code. But then I renamed the class back how it was, so the change above isn't needed. I'll put it back, agree with you the original was clearer